os-eap-probes/added/livenessProbe.sh

```diff
@@ -23,11 +24,25 @@ if [ $# -gt 0 ] ; then
     COUNT=$1
 fi
 
-# setting the first parameter to true/false, gets you the new default behaviour of
+# check if this is OCP 4 (k8s 13+)
+# avoid querying the api every time if we can.
+is_ocp4="false"
+if [ -n "$KUBERNETES_SERVICE_HOST" ]; then
+    if [ -f ${OCPVERS} ]; then
+        is_ocp4=$(<${OCPVERS})
+    else
+        is_ocp4=$(is_ocp4_or_greater)
+        echo ${is_ocp4} > ${OCPVERS}
+    fi
+fi
+
+# setting the first parameter to true/false, gets you the new default behavior of
 # count = 1, sleep = 0 and initial sleep seconds = 0
-# setting it to a numeric value, or nothing falls through to the old behaviour
+# setting it to a numeric value, or nothing falls through to the old behavior
+# if this is OCP 4 and it is unset, default to the new behavior
 
-if [[ -n "$COUNT" && ( $COUNT = "true" || $COUNT = "false" ) ]]; then
+# COUNT is unset, and we're running on OCP 4+ OR COUNT is set to either true or false
+if [[ ( $# -eq 0 && ${is_ocp4} = "true" ) || ( -n ${COUNT} && ( ${COUNT} = "true" || ${COUNT} = "false" )) ]]; then
```

Comment:
Is it possible to get rid of
```
... || ( -n ${COUNT} && ( ${COUNT} = "true" || ${COUNT} = "false" ))
```
in the if statement? I'm not sure for which purpose this might be needed.

Reply:
@mnovak1 This is there to support the true / false parameter (even on OCP 3.x) if you want to use the updated probe logic.

Reply:
Ok, got it. I think there nothing what this might break.